method-scripts-berterniereport.xml

```diff
@@ -1,16 +1,25 @@
 <es:transforms xmlns:es="urn:hpcc:esdl:script">
   <es:transform name="BertErnieReportOutOfBand" target="soap:Body/{$query}">
-    <es:choose>
-      <es:when test="true()">
-        <es:set-value target="_LogFunctionName" value="&apos;BertErnieRpt&apos;"/>
-      </es:when>
-    </es:choose>
+    <!--
+    The log-as function name is the epitome of method-specific.
+
+    It can be replaced by a shared script if the shared script contains a conditional for
+    every method. However, some methods change the value based on request content, so a
+    different method might have a muvh more complex implementation.
+
+    It could be replaced in many cases by method-specific external variable or input
+    definitions.
+    -->
+    <es:set-value target="_LogFunctionName" value="&apos;BertErnieRpt&apos;"/>
   </es:transform>
-  <es:transform name="BertErnieReportInBand" target="soap:Body/{$query}/{$request}" xmlns:es="hpcc-crt">
-    <es:choose>
-      <es:when test="$internal_user=0 and $test_user=1">
-        <es:append-to-value target="Row/Options/IsOnline" value="&apos;1&apos;"/>
-      </es:when>
-    </es:choose>
+  <es:transform name="BertErnieReportInBand" target="soap:Body/{$query}/{$request}">
+    <!--
+    This applies to only the report method, although as the poriginal author I do know
+    that the first part of the test would make transform safe to be shared with all
+    service methods except OscarSearch.
+    -->
+    <es:if test="$internal_user=0 and $test_user=1">
+      <es:append-to-value target="Row/Options/IsOnline" value="&apos;1&apos;"/>
```

Comment:
Why apped-to here?

Reply:
Because I missed it. There is no case where I wanted to use append-to-value; I had to when I didn't know if the target path existed.